packages/wonder-blocks-dropdown/components/dropdown-core.js

```diff
@@ -65,22 +79,59 @@ export default class DropdownCore extends React.Component<DropdownCoreProps> {
     }
 
     componentDidMount() {
+        this.updateEventListeners();
+    }
+
+    componentDidUpdate(prevProps: DropdownCoreProps) {
+        if (prevProps.open !== this.props.open) {
+            this.updateEventListeners();
+        }
+
+        // If the menu just changed from closed to open, and if we haven't
+        // found the correct menuWidth yet. Only applies to right aligned.
```

Comment:
I'm not enthused by this, but at least we only need it for this one case.

Reply:
Turns out I was wrongly positioning something using `absolute` when I no longer needed to! So I was able to remove all this calculation stuff. 😁